### ISSUE TITLE
Python version selection

### ISSUE THREAD

**Issue by eirnym** [macvim-dev/macvim#107]
Are there option to select which python to load?

MacVim: Snapshot 79

**Comment by splhack** [comment]
tl;dr. Not yet.


Snapshot 79 Vim executable was built with Python 2.7 and Python 3.5. It means Vim only can load Python 2.7 and Python 3.5 shared libraries (dylib). And, in the current source code, Vim only can load the shared libraries from the very specific path what was defined by configure script. If you want to use your own Python 2.7 and Python 3.5 builds like MacPorts, please wait the patch that I mentioned in #106. Also Python interface binary compatibility is limited, for example, you can't use Python 2.6 and Python 3.4. In addition, Vim can't load Python 3.5 after Python 2.7 loaded, Vim can't load Python 2.7 after Python 3.5 loaded as well.

**Comment by eirnym** [comment]
Can this path be simlinked? 

What does this patch give for python linking? I don't see anything matching.

**Comment by splhack** [comment]
yeah (I guess)

idk, I think you can use opensnoop to find out the path.

**Comment by splhack** [comment]
For Python 2.7, IIRC, Vim can load the framework from the system.

For Python 3.5, for example, with Homebrew python3, you may be able to use :python3 with the symbolic link.

    $ ln -s /usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework ~/Library/Frameworks

**Comment by splhack** [comment]
You can now use `pythondll` and `python3dll` option to specify Python shared library.
https://github.com/vim/vim/commit/d94464ee294a351ce7b6ba18e8bd3f24f1bef920
